odoo/addons/test_lint/tests/test_manifests.py

```diff
@@ -58,6 +59,27 @@ def _test_manifest_values(self, module, manifest_data):
             _logger.warning(
                 "Module %r specific to one single country %r should contain `l10n` in their name.",
                 module, manifest_data['countries'][0])
+        
+        # Modules with a countries key in the manifest will only be auto-installed
+        # in case a company with a matching country exists.
+        # See https://github.com/odoo/odoo/pull/144586
+        countries = manifest_data.get("countries")
+        depends = manifest_data.get("depends", [])
+        auto_install = manifest_data.get("auto_install")
+
+        has_l10n_dependency = any(re.match(r"^l10n_[a-z]{2}$", dep) for dep in depends)
+        is_auto_install_bool = isinstance(auto_install, bool) and auto_install
+        is_auto_install_list = isinstance(auto_install, list) and any(
+            re.match(r"^l10n_[a-z]{2}$", item) for item in auto_install
```

Comment:
It doesn't have to be that exact regex.
For instance, a module depending on `l10n_xx_reports` or `l10n_xx_edi` should not have the `countries` in the manifest either